### ISSUE TITLE
`Lint/UselessMethodDefinition` should ignore required keyword arguments

### ISSUE THREAD

**Issue by fynsta** [rubocop/rubocop#13013]
Not sure if this is reasonable, but I think `Lint/UselessMethodDefinition` should ignore required keyword arguments.

Consider the following code:
```rb
class Parent
  def initialize(required_arg: nil)
     ...
  end
end

class Child < Parent
  def initialize(required_arg:)
    super
  end
end
```

Rubocop wants me to delete this method, but that would completely change the behaviour (i.e. not throw an error when the argument is missing).

## Expected behavior

Not to register an offense.

## Actual behavior

Rubocop registers a `Lint/UselessMethodDefinition` offense.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.64.1 (using Parser 3.3.3.0, rubocop-ast 1.31.3, running on ruby 3.2.4) [arm64-darwin23]
```

**Comment by Earlopain** [comment]
This behaviour is described in the cops safety section and expected: https://docs.rubocop.org/rubocop/cops_lint.html#lintuselessmethoddefinition

> This cop is unsafe as it can register false positives for cases when an empty constructor just overrides the parent constructor, which is bad anyway.

You can disable the cop just for this line through a disable comment.

**Comment by fynsta** [comment]
Ah got it! Sorry, I read that comment as "empty parameter list", but that makes more sense!

**Comment by Earlopain** [comment]
I think that sentence can be improved. For example, it doesn't only apply to constructors but any method (the "constructor" is just a regular method anyways). The specs also make it look like there would be no offense for the inverted case where a method with optional parameters only calls `super` (like in your parent class).
